giraffe/src/utils/tableGraph.ts

```diff
@@ -0,0 +1,368 @@
+  const ignoredIndices = IGNORED_COLUMNS.map(name => header.indexOf(name))
+
+  const excludedData = data.map(row => {
+    return row.filter((__, i) => !ignoredIndices.includes(i))
+  })
+
+  return excludedData
+}
+
+export const transformTableData = (
+  data: string[][],
+  sortOptions: SortOptions,
+  renamableFields: RenamableField[],
+  tableOptions: TableOptions,
+  timeFormat: string,
+  decimalPlaces: DecimalPlaces
+): TransformTableDataReturnType => {
+  const {verticalTimeAxis = DEFAULT_VERTICAL_TIME_AXIS} = tableOptions
+
+  const excludedData = excludeNoisyColumns(data)
+
+  const resolvedRenamableFields = resolveRenamableFields(
+    renamableFields,
+    excludedData[0]
+  )
+
+  const {sortedData, sortedTimeVals} = sortTableData(excludedData, sortOptions)
+
+  const filteredData = filterTableColumns(sortedData, resolvedRenamableFields)
+
+  const orderedData = orderTableColumns(filteredData, resolvedRenamableFields)
+
+  const transformedData = verticalTimeAxis ? orderedData : unzip(orderedData)
+
+  const columnWidths = calculateColumnWidths(
+    transformedData,
+    resolvedRenamableFields,
+    timeFormat,
+    verticalTimeAxis,
+    decimalPlaces
+  )
+
+  return {
+    transformedData,
+    sortedTimeVals,
+    columnWidths,
+    resolvedRenamableFields,
+    sortOptions,
+  }
+}
+
+/*
+  Checks whether an input value of arbitrary type can be parsed into a
+  number. Note that there are two different `isNaN` checks, since
+
+  - `Number('')` is 0
+  - `Number('02abc')` is NaN
+  - `parseFloat('')` is NaN
+  - `parseFloat('02abc')` is 2
+
+*/
+export const isNumerical = (x: any): boolean =>
+  !isNaN(Number(x)) && !isNaN(parseFloat(x))
+
+export const findHoverTimeIndex = (
+  sortedTimeVals: string[],
+  hoverTime: number
+) => {
+  if (sortedTimeVals.length < 2) {
+    // first value is "_time" header
+    return -1
+  }
+
+  const firstDiff = getUnixISODiff(hoverTime, sortedTimeVals[1]) // sortedTimeVals[0] is "_time"
+  const hoverTimeFound = fastReduce<string, {index: number; diff: number}>(
+    sortedTimeVals,
+    (acc, currentTime, index) => {
+      const thisDiff = getUnixISODiff(hoverTime, currentTime)
+      if (thisDiff < acc.diff) {
+        return {index, diff: thisDiff}
+      }
+      return acc
+    },
+    {index: 1, diff: firstDiff}
+  )
+
+  return hoverTimeFound.index
+}
+
+/**
+ * Get absolute mili seconds between a unix ms time number and iso utc string
+ *
+ * @param unixMs
+ * @param isoTime
+ */
+export const getUnixISODiff = (unixMs: number, isoTime: string | number) => {
+  return Math.abs(unixMs - new Date(isoTime).valueOf())
+}
+
+export const findTableNameHeaders = (tables: FluxTable[], name: string) => {
+  const foundTable = tables.find(t => t.name === name)
+
+  return get(foundTable, 'data.0', [])
+}
+
+export const resolveTimeFormat = (timeFormat: string) => {
+  if (FORMAT_OPTIONS.find(d => d.text === timeFormat)) {
+    return timeFormat
+  }
+
+  return DEFAULT_TIME_FORMAT
+}
```

Comment:
This is another file i'd like to flag as being potentially being performance heavy due to the amount of looping

Reply:
Agreed. We should explore this in another story. Right now, we just want to get Tables into Giraffe.